site/blog/data-poisoning.md

```diff
@@ -0,0 +1,118 @@
+
+### Monitor Model Behavior to Identify Data Poisoning
+
+Regularly monitor the outputs of your LLM for signs of unusual or undesirable behavior.
+
+- **Implement Tracing**: LLM tracing provides a detailed snapshot of the decision-making and thought processes within LLMs as they generate responses. Tracing can help you monitor, debug, and understand the execution of an LLM application
+- **Use Golden Datasets**: Golden datasets in LLMs are high-quality, carefully curated collections of data used to evaluate and benchmark the performance of large language models. Use these datasets as a "ground truth" to evaluate the performance of your models.
+- **Test with Adversarial Examples**: [Use Promptfoo](https://www.promptfoo.dev/docs/red-team/quickstart/) to test your models with adversarial inputs to evaluate its robustness against potential attacks.
+- **Deploy Guardrails**: Use guardrails as a defense-in-depth measure to prevent the LLM from generating outputs that violate your policies.
+
+### Limit Access to Training Processes to Prevent Data Poisoning
+
+Restrict who can modify training data or initiate training processes.
+
+- **Lock Down Access**: Restrict access to LLM repositories and implement robust monitoring to mitigate the risk of insider threats.
+  - Access to training data should be restricted based on least privilege and need-to-know. Access should be recertified on a regular cadence (such as quarterly) to account for employee turnover or job changes.
+  - All access should be logged and audited. Developer access should be limited to the minimum necessary to perform their job and access should be revoked when they leave the organization.
+- **Audit Logs**: Keep detailed logs of data access and modifications to trace any unauthorized activities connected to your training data or LLM configurations.
+
+### Enforce Supply Chain Security to Identify Poisoned Models
+
+- **Vet Your Sources**: Conduct thorough due diligence on model providers and training data sources.
+  - Review model cards and documentation to understand the model's training processes and performance. You can learn more about this in our [foundation model security](https://www.promptfoo.dev/blog/foundation-model-security/) blog post.
+  - Verify that models downloaded from Hugging Face [pass their malware scans](https://huggingface.co/docs/hub/en/security-malware) and [pickling scans](https://huggingface.co/docs/hub/en/security-pickle).
+
+### Red Team LLM Applications to Detect Data Poisoning
+
+- **Model Red Teaming**: Run an initial [red team](https://www.promptfoo.dev/docs/red-team/) assessment against any models pulled from shared or public repositories like Hugging Face.
+- **Assess Bias**: In Promptfoo's eval framework, use Promptfoo's [classifier assert type](https://www.promptfoo.dev/docs/configuration/expected-outputs/classifier/#bias-detection-example) to assess grounding, factuality, and bias in models pulled from Hugging Face.
```

Comment:
not just related to hugging face this can be run on any output. model is hosted on hugging face